src/loaders/ncbi_downloader/gtdb.py

```diff
@@ -0,0 +1,287 @@
+        data_dir = os.path.join(work_dir, genome_id)
+        if not os.path.exists(data_dir):
+            genome_ids.append(genome_id)
+            continue
+
+        ext_count = 0
+        for file_name in os.listdir(data_dir):
+            if any([file_name.endswith(ext) for ext in target_file_ext]) and all(
+                [substring not in file_name for substring in exclude_name_substring]
+            ):
+                ext_count += 1
+
+        if ext_count != target_ext_count:
+            genome_ids.append(genome_id)
+
+    return genome_ids
+
+
+def _get_parser():
+
+    parser = argparse.ArgumentParser(
+        description="PROTOTYPE - Download genome files from NCBI FTP server.",
+        formatter_class=loader_helper.ExplicitDefaultsHelpFormatter,
+    )
+
+    gtdb_release_vers = _parse_gtdb_release_vers()  # available GTDB release versions
+
+    required = parser.add_argument_group("required named arguments")
+    optional = parser.add_argument_group("optional arguments")
+
+    # Required flag argument
+    required.add_argument(
+        "--release_ver",
+        required=True,
+        type=str,
+        choices=gtdb_release_vers,
+        help="GTDB release version",
+    )
+
+    # Optional argument
+    optional.add_argument(
+        "--root_dir",
+        type=str,
+        default=loader_common_names.ROOT_DIR,
+        help="Root directory",
+    )
+    optional.add_argument(
+        "--download_file_ext",
+        type=str,
+        nargs="+",
+        default=DOWNLOAD_FILE_EXT,
+        help="Download only files that match given extensions",
+    )
+    optional.add_argument("--threads", type=int, help="Number of threads")
+    optional.add_argument(
+        "--overwrite", action="store_true", help="Overwrite existing files"
+    )
+    optional.add_argument(
+        "--exclude_name_substring",
+        type=str,
+        nargs="+",
+        default=loader_common_names.STANDARD_FILE_EXCLUDE_SUBSTRINGS,
+        help="Files with a specific substring in their names that should be excluded from the download",
+    )
+    return parser
+
+
+def main():
+    parser = _get_parser()
+    args = parser.parse_args()
+
+    release_ver = args.release_ver
+    root_dir = args.root_dir
+    threads = args.threads
+    overwrite = args.overwrite
+    download_file_ext = args.download_file_ext
+    exclude_name_substring = args.exclude_name_substring
+
+    if threads and (threads < 1 or threads > cpu_count()):
+        parser.error(f"minimum thread is 1 and maximum thread is {cpu_count()}")
+
+    work_dir = _make_work_dir(root_dir)
+    csd = loader_helper.make_collection_source_dir(
+        root_dir,
+        loader_common_names.DEFAULT_ENV,
+        KBASE_COLLECTION,
+        release_ver,
+    )
+    genome_ids_unprocessed, taxonomy_urls = _fetch_gtdb_genome_ids(release_ver, work_dir)
+    taxonomy_files = [os.path.basename(url) for url in taxonomy_urls]
+    genome_ids = _process_genome_ids(
```

Comment:
Now I look at it, _process_genome_ids is non-gtdb specific as well and should be moved into the ncbi code for reuse, probably with a better name, maybe something like `remove_ids_with_existing_data` or something

Reply:
Updated!

Reply:
:+1: